lib/posix/pthread_rwlock.c

```diff
@@ -25,6 +25,9 @@ static u32_t write_lock_acquire(pthread_rwlock_t *rwlock, s32_t timeout);
 int pthread_rwlock_init(pthread_rwlock_t *rwlock,
 			const pthread_rwlockattr_t *attr)
 {
+	if (rwlock == NULL)
+		return ENOMEM;
```

Comment:
@punitvara How do you say that there is insufficient memory (that is what ENOMEM indicates) to initialize rwlock when it is NULL? I am not clear on this part. Please clarify.

Reply:
> How do you say that there is insufficient memory (that is what ENOMEM indicates) to initialize rwlock when it is NULL? I am not clear on this part. Please clarify.

Why do you think that rwlock being NULL means that there is insufficient memory?

Reply:
@pfalcon : That is exactly what I am trying to understand. Why should we return ENOMEM when rwlock is NULL?

Reply:
@spoorthik: Well, right, my comment would rather be addressed to @punitvara, as the author of the patch, even if it reuses your words ;-). Sorry for the confusion!